From 8e882df78ede98c15a3f2567fabebfde1d774b02 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex@linutronix.de>
Date: Fri, 7 Jan 2022 21:20:15 +0100
Subject: [PATCH] src/fundamental: list fundamental_source_paths using relative
 paths

Otherwise, the compiler takes the full path to the source file
and writes it into the binary output, breaking reproducibility.

Upstream-Status: Submitted [https://github.com/systemd/systemd/pull/22047]
Signed-off-by: Alexander Kanavin <alex@linutronix.de>
---
 src/fundamental/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/fundamental/meson.build b/src/fundamental/meson.build
index 287f0fe36a..954bdf150b 100644
--- a/src/fundamental/meson.build
+++ b/src/fundamental/meson.build
@@ -20,7 +20,7 @@ sources = '''
 # for sd-boot
 fundamental_source_paths = []
 foreach source : sources
-        fundamental_source_paths += meson.current_source_dir() / source
+        fundamental_source_paths += '../../fundamental/' + source
 endforeach
 
 # for libbasic
